src/main/java/org/truffleruby/core/string/StringNodes.java

```diff
@@ -3553,20 +3553,16 @@ public int stringCharacterByteIndexMultiByteEncoding(DynamicObject string, int c
 
         @Specialization(guards = "isSingleByteOptimizable(string)")
         public int singleByte(DynamicObject string, int byteIndex) {
-            // Taken from Rubinius's String::find_byte_character_index.
             return byteIndex;
         }
 
         @Specialization(guards = { "!isSingleByteOptimizable(string)", "isFixedWidthEncoding(string)" })
         public int fixedWidth(DynamicObject string, int byteIndex) {
-            // Taken from Rubinius's String::find_byte_character_index.
             return byteIndex / encoding(string).minLength();
         }
 
         @Specialization(guards = { "!isSingleByteOptimizable(string)", "!isFixedWidthEncoding(string)", "isValidUtf8(string)" })
         public int validUtf8(DynamicObject string, int byteIndex) {
-            // Taken from Rubinius's String::find_byte_character_index.
```

Comment:
I'd rather these comments stay here. They help cross-reference to the original source.

Reply:
But they are useless here, the code is straighforward.
Also, there is already one comment below:
```
// Taken from Rubinius's String::find_byte_character_index and Encoding::find_byte_character_index.
```
In general, I'd rather drop all comments like that and make the code our own and clean. We are not going to re-translate code from C++.
Documentation of StringSupport methods would be much more valuable imho.

Reply:
If @nirvdrum is still actively working on this code more than most people and he uses the comments, then I think it's reasonable to leave them in.

Reply:
The one I linked to is directly above a TODO comment about code that hasn't yet been translated. The others could maybe go since their implementation is the most straightforward and obvious solution anyway.

Reply:
OK, I'll add this one back.